### PR TITLE
Add support for 1.17 protocol

### PR DESCRIPTION
**Pull request by nickelpro**
Diff against last supported pre-release viewable [here](https://www.jstoolset.com/diff/80240b9fa326ba27)

Packet layout based on the Pre-Release documentation and Burger

### CODE REVIEW

**Comment by rom1504** [comment]
Could you do one commit with the unchanged protocol and one commit with the changes ?
That makes reviewing possible

**Comment by nickelpro** [comment]
It dirties the commit history, a diff of the protocols is available in my parent comment

**Comment by rom1504** [comment]
I will squash afterwards but having these 2 commits makes it possible to understand when looking at GitHub ui diff

**Comment by nickelpro** [comment]
Rebased for version diff

**Comment by rom1504** [comment]
just merged the data pr, could you rebase ?

**Comment by nickelpro** [comment]
I mean, somewhat pointless to do so since in merging data the protocol has already been added

**Comment by rom1504** [comment]
oh I didn't notice he took your commit

**Comment by rom1504** [comment]
ok so anyway I checked the diff and it seems mostly correct, there was a lot of stuff changed, didn't check everything

**Comment by rom1504** [comment]
we'll catch potential issues downstream I guess

File: data/pc/1.17/protocol.json [hunk 1]
```diff
@@ -4198,6 +3973,162 @@
...
+              "name": "warningTime",
+              "type": "varint"
+            }
+          ]
+        ],
+        "packet_world_border_warning_reach": [
+          "container",
+          [
+            {
+              "name": "warningBlocks",
```
Comment (rom1504) [inline data/pc/1.17/protocol.json:4083]: Remove the s

File: data/pc/1.17/protocol.json [hunk 2]
```diff
@@ -2956,18 +2929,12 @@
             }
           ]
         ],
-        "packet_entity_destroy": [
+        "packet_destroy_entity": [
           "container",
           [
             {
               "name": "entityIds",
```
Comment (rom1504) [inline data/pc/1.17/protocol.json:2936]: Remove the s

**Pull request closed without merging**